Description: Make it possible to compile evince 3.8 against poppler 0.18
 Since poppler 0.20.0 is not yet available in Debian sid, add a version
 check so we can build against older versions. The existing code already
 provides a sensible fallback path.
Author: Michael Biebl <biebl@debian.org>
--- a/backend/pdf/ev-poppler.cc
+++ b/backend/pdf/ev-poppler.cc
@@ -1115,10 +1115,10 @@
 		const char *type_str;
 		const char *embedded;
 		const char *standard_str = "";
-		const gchar *substitute;
+		const gchar *substitute = NULL;
 		const gchar *substitute_text;
 		const gchar *filename;
-		const gchar *encoding;
+		const gchar *encoding = NULL;
 		const gchar *encoding_text;
 		char *details;
 		
@@ -1128,7 +1128,9 @@
 			name = _("No name");
 		}
 
+#if POPPLER_CHECK_VERSION(0, 20, 0)
 		encoding = poppler_fonts_iter_get_encoding (iter);
+#endif
 		if (!encoding) {
 			/* translators: When a font type does not have
 			   encoding information or it is unknown.  Example:
@@ -1165,7 +1167,9 @@
 			}
 		}
 
+#if POPPLER_CHECK_VERSION(0, 20, 0)
 		substitute = poppler_fonts_iter_get_substitute_name (iter);
+#endif
 		filename = poppler_fonts_iter_get_file_name (iter);
 		encoding_text = _("Encoding");
 		substitute_text = _("Substituting with");
--- a/configure.ac
+++ b/configure.ac
@@ -474,7 +474,7 @@
   [enable_pdf=yes])
 
 if test "x$enable_pdf" = "xyes"; then
-    POPPLER_REQUIRED=0.20.0
+    POPPLER_REQUIRED=0.18.0
     PKG_CHECK_MODULES(POPPLER, poppler-glib >= $POPPLER_REQUIRED libxml-2.0 >= $LIBXML_REQUIRED,enable_pdf=yes,enable_pdf=no)
 
     if test "x$enable_pdf" = "xyes"; then
